ringhold/src/intTest/java/org/opencadc/ringhold/InventoryValidatorTest.java

```diff
@@ -0,0 +1,328 @@
+            if (opt.exists()) {
+                Properties props = new Properties();
+                props.load(new FileReader(opt));
+
+                if (props.containsKey("inventoryServer")) {
+                    INVENTORY_SERVER = props.getProperty("inventoryServer").trim();
+                }
+                if (props.containsKey("inventoryDatabase")) {
+                    INVENTORY_DATABASE = props.getProperty("inventoryDatabase").trim();
+                }
+                if (props.containsKey("inventorySchema")) {
+                    INVENTORY_SCHEMA = props.getProperty("inventorySchema").trim();
+                }
+            }
+        } catch (MissingResourceException | FileNotFoundException noFileException) {
+            log.debug("No intTest.properties supplied.  Using defaults.");
+        } catch (IOException oops) {
+            throw new RuntimeException(oops.getMessage(), oops);
+        }
+
+        log.debug("intTest database config: " + INVENTORY_SERVER + " " + INVENTORY_DATABASE + " "
+                      + INVENTORY_SCHEMA);
+    }
+
+    final ArtifactDAO artifactDAO;
+    final DeletedStorageLocationEventDAO deletedStorageLocationEventDAO;
+    final Map<String, Object> daoConfig = new TreeMap<>();
+    final String jndiPath = "jdbc/InventoryEnvironment";
+
+    public InventoryValidatorTest() throws Exception {
+        artifactDAO = new ArtifactDAO();
+        deletedStorageLocationEventDAO = new DeletedStorageLocationEventDAO();
+
+        final DBConfig dbrc = new DBConfig();
+        final ConnectionConfig connectionConfig = dbrc.getConnectionConfig(INVENTORY_SERVER, INVENTORY_DATABASE);
+        DBUtil.createJNDIDataSource(jndiPath, connectionConfig);
+
+        daoConfig.put(SQLGenerator.class.getName(), SQLGenerator.class);
+        daoConfig.put("jndiDataSourceName", jndiPath);
+        daoConfig.put("database", INVENTORY_DATABASE);
+        daoConfig.put("schema", INVENTORY_SCHEMA);
+
+        artifactDAO.setConfig(daoConfig);
+        deletedStorageLocationEventDAO.setConfig(daoConfig);
+    }
+
+    @Before
+    public void setup() throws Exception {
+        truncateTables();
+    }
+
+    @After
+    public void cleanup() throws Exception {
+        truncateTables();
+    }
+
+    @Test
+    public void noArtifactsMatchFilter() throws Exception {
+        writeConfig();
+
+        Artifact a1 = getTestArtifact("cadc:TEST/one.txt");
+        this.artifactDAO.put(a1);
+        Artifact a2 = getTestArtifact("cadc:INT/two.txt");
+        this.artifactDAO.put(a2);
+        Artifact a3 = getTestArtifact("cadc:CADC/three.txt");
+        this.artifactDAO.put(a3);
+
+        InventoryValidator testSubject = new InventoryValidator(this.daoConfig, this.daoConfig);
+        testSubject.run();
+
+        ResourceIterator<Artifact> artifacts = this.artifactDAO.iterator("uri LIKE 'cadc:INTTEST/%'", null);
+        Assert.assertFalse(artifacts.hasNext());
+
```

Comment:
The above iterator and assert look like a re-run of what happens inside the InventoryHarvester... never like to see implementation code repeated in the tests. The checks below assert that the result is correct.

Comment applies to other test methods as well.

Reply:
No it's not. It was checking if there are any results, not specific results, but the following checks do that too. Removed.